src/python/WMCore/WMSpec/WMTask.py

```diff
@@ -537,22 +537,9 @@ def setJobResourceInformation(self, timePerEvent=None, sizePerEvent=None, memory
             performanceParams.timePerEvent = timePerEvent or getattr(performanceParams, "timePerEvent")
         if sizePerEvent or getattr(performanceParams, "sizePerEvent", None):
             performanceParams.sizePerEvent = sizePerEvent or getattr(performanceParams, "sizePerEvent")
-
-        # special handling for memory overwrite during assignment
-        if isinstance(memoryReq, dict):
-            taskMemory = memoryReq.get(self.name())
-        else:
-            taskMemory = memoryReq
-
```

Comment:
Also add for timePerEvent and sizePerEvents as well if those need to be overwritten?